src/main/java/spoon/support/reflect/reference/CtTypeParameterReferenceImpl.java

```diff
@@ -204,12 +204,21 @@ public CtTypeParameter getDeclaration() {
 
 		CtElement e = this;
 		CtElement parent = getParent();
+
 		if (parent instanceof CtTypeReference) {
-			if (parent.isParentInitialized() == false) {
-				return null;
+			if (!parent.isParentInitialized()) {
+				// we might enter in that case because of a call
+				// of getSuperInterfaces() for example
```

Comment:
This case should not happen after my change in `CtTypeReference#getSuperInterfaces`. Let's rollback this change. WDYT?

Reply:
I just checked and I have the following broken test if I remove it: GenericsTest#testTypeParameterReferenceAsActualTypeArgument.